internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,206 @@
+//nolint:deprecatedComment
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+const (
+	sqlMIServiceName   = "SQL Managed Instance"
+	sqlMIProductFamily = "Databases"
+)
+
+// SQLManagedInstance struct represents an azure Sql Managed Instance.
+//
+// The azurerm_sql_managed_instance resource is deprecated in version 3.0 of the AzureRM provider and will be removed in version 4.0.
+// Please use the azurerm_mssql_managed_instance resource instead when available in infracost
+//
+// Only support for Gen5 is available on that resource
```

Comment:
```suggestion
// SQLManagedInstance currently only supports Gen5 database instances.
```

Reply:
this should fix linting, and we don't really need the deprecation version comment here.